### PR TITLE
[Terrain] Fix the clipmap rendering when macro normal map is not set

### PR DESCRIPTION
**Pull request by jiaweig-amzn**
Encoded the "has normal" info into the macro color alpha channel, so that in the forward pass it knows when to use the vertex normal instead.

Signed-off-by: jiaweig <51759646+jiaweig-amzn@users.noreply.github.com>

### CODE REVIEW

**Review by mbalfour-amzn — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli [hunk 1]
```diff
@@ -399,9 +400,11 @@ ClipmapSample SampleClipmap(float2 worldPosition)
...
         );
 
-        data.m_macroColor = ColorBilinearSampling(
+        float4 macroColor = ColorBilinearSampling(
             TerrainSrg::m_macroColorClipmaps, macroClipmapUvs, macroClipmapLevel.m_closestLevel,
-            TerrainSrg::m_clipmapData.m_enableMacroClipmapOverlay).rgb;
+            TerrainSrg::m_clipmapData.m_enableMacroClipmapOverlay);
+        data.m_macroColor = macroColor.rgb;
+        data.m_hasMacroNormal = macroColor.a == 1.0;
         data.m_macroNormal = NormalBilinearSampling(TerrainSrg::m_macroNormalClipmaps, macroClipmapUvs, macroClipmapLevel.m_closestLevel);
```
Comment (invertednormal) [inline Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli:408]: nit: might be worth not calling `NormalBilinearSampling()` when hasMacroNormal is false. I'd expect all threads take the same path for this most of the time.

Comment (invertednormal) [reply Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli:408]: Same for below

File: Gems/Terrain/Assets/Shaders/Terrain/TerrainMacroClipmapGenerationPass.azsl [hunk 2]
```diff
@@ -57,9 +59,10 @@ void MainCS(
 
             float3 macroColor = TerrainMaterialSrg::m_baseColor.rgb;
             float3 macroNormal = float3(0.0, 0.0, 1.0);
-            SampleMacroTexture(worldPosition, positionDdx, positionDdy, macroColor, macroNormal);
+            bool hasNormal;
+            SampleMacroTexture(worldPosition, positionDdx, positionDdy, macroColor, macroNormal, hasNormal);
 
-            PassSrg::m_macroColorClipmaps[texelIndex] = float4(macroColor, 1.0);
+            PassSrg::m_macroColorClipmaps[texelIndex] = float4(macroColor, hasNormal ? HAS_NORMAL : NO_NORMAL);
```
Comment (invertednormal) [inline Gems/Terrain/Assets/Shaders/Terrain/TerrainMacroClipmapGenerationPass.azsl:65]: If you're going to use the macros NO_NORMAL and HAS_NORMAL here, you should probably use them in ClipmapComputeHelpers.azsli too. ie, `data.m_hasMacroNormal = macroColor.a == HAS_NORMAL;`

**Review by invertednormal — APPROVED** [review]
Looks good, just a few suggestions but nothing blocking.

**Pull request merged**